File: BehaviorPattern/Chain_Of_Responsibility/Java/Concretes/SupportRequest.java
package Concretes;

public class SupportRequest {
    private String message;
    private int complexity;

    public SupportRequest(String message, int complexity) {
        this.message = message;
        this.complexity = complexity;
    }

    public String getMessage() {
        return message;
    }

    public int getComplexity() {
        return complexity;
    }
}

File: BehaviorPattern/Chain_Of_Responsibility/Golang/SupportRequest.go
package main

type SupportRequest struct {
    Message    string
    Complexity int
}

func NewSupportRequest(message string, complexity int) *SupportRequest {
    return &SupportRequest{
        Message:    message,
        Complexity: complexity,
    }
}

func (sr *SupportRequest) GetMessage() string {
    return sr.Message
}

func (sr *SupportRequest) GetComplexity() int {
    return sr.Complexity
}
